src/app/Fake.Windows.Wix/Wix.fs

```diff
@@ -0,0 +1,2008 @@
+    /// on component reference counts to be accurate during installation of the upgrade product and removal of the installed product. For more information, see Bob Arnson's blog post 
+    /// "Paying for Upgrades" for details. If installation of the upgrade product fails, Windows Installer also rolls back the removal of the installed product -- in other words, reinstalls it. 
+    | AfterInstallExecute
+    /// Schedules RemoveExistingProducts between the InstallExecuteAgain and InstallFinalize standard actions. 
+    /// This is identical to the afterInstallExecute scheduling but after the InstallExecuteAgain standard action instead of InstallExecute. 
+    | AfterInstallExecuteAgain
+    /// Schedules RemoveExistingProducts after the InstallFinalize standard action. This is similar to the afterInstallExecute and afterInstallExecuteAgain schedulings but takes place outside 
+    /// the installation transaction so if installation of the upgrade product fails, Windows Installer does not roll back the removal of the installed product, 
+    /// so the machine will have both versions installed. 
+    | AfterInstallFinalize
+    override m.ToString() =
+        match m with
+        | AfterInstallValidate -> "afterInstallValidate"
+        | AfterInstallInitialize -> "afterInstallInitialize"
+        | AfterInstallExecute -> "afterInstallExecute"
+        | AfterInstallExecuteAgain -> "afterInstallExecuteAgain"
+        | AfterInstallFinalize -> "afterInstallFinalize"
+
+/// Parameters for WiX Major Upgrade
+type MajorUpgrade = 
+    {
+        /// Determines the scheduling of the RemoveExistingProducts standard action, which is when the installed product is removed. The default is "afterInstallValidate" which removes the 
+        /// installed product entirely before installing the upgrade product. It's slowest but gives the most flexibility in changing components and features in the upgrade product.
+        Schedule : MajorUpgradeSchedule
+        /// When set to no (the default), products with lower version numbers are blocked from installing when a product with a higher version is installed; the DowngradeErrorMessage 
+        /// attribute must also be specified. When set to yes, any version can be installed over any other version. 	 
+        AllowDowngrades : YesOrNo
+        /// The message displayed if users try to install a product with a lower version number when a product with a higher version is installed. Used only when AllowDowngrades is no (the default). 
+        DowngradeErrorMessage : string
+    }
+    override w.ToString() =
+        let downgradeErrorMessage =
+            match w.AllowDowngrades with
+            | Yes -> ""
+            | No ->  " DowngradeErrorMessage=\"" + w.DowngradeErrorMessage + "\""
+        "<MajorUpgrade Schedule=\"" + w.Schedule.ToString() + 
```

Comment:
remove this function?

Reply:
I don't quite understand, which function I should remove here. Can you please clarify?